keps/sig-etcd/4331-livez-readyz/README.md

```diff
@@ -0,0 +1,116 @@
+# KEP-4331: Livez and Readyz Probes
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+- [Definition for livez and readyz](#definition-for-livez-and-readyz)
+- [Design Details](#design-details)
+  - [API Design](#api-design)
+  - [Failure Modes and Detection Methods](#failure-modes-and-detection-methods)
+    - [Detection](#detection)
+- [Implementation Plan](#implementation-plan)
+- [Future Discussions/Improvements](#future-discussionsimprovements)
+<!-- /toc -->
+
+## Summary
+This is the KEP of the original design doc of [etcd livez and readyz probes](https://docs.google.com/document/d/1PaUAp76j1X92h3jZF47m32oVlR8Y-p-arB5XOB7Nb6U)
+
+## Motivation
+The current etcd implementation has a single `/health` probe that is used to determine both the liveness and readiness of a node.
+
+What does it do?
+* check if local node has leader (detects network partition)
+* check if local node has any alarm activated
+* check if local node is capable of serving a linearizable read request within hard-coded timeout (5s + 2 * election-timeout) (default 7s)
+
+The current health probe is not Kubernetes API compliant. 
+It does not properly reflect whether etcd needs to restart or stop taking traffic. etcd liveness and readiness probe configured with kubeadm using the same health probe is incorrect. 
```

Comment:
I wouldn't say the existing kubeadm's usage on the /health endpoint is incorrect. 
- The new /livez and /readyz endpoints are clearer and easy to understand & follow.
- Of course, the new KEP also enhances the capabilities (e.g add the detection of raft loop deadlock).

Reply:
thanks for the feedback. Readjusted the wording.